compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -137,33 +137,161 @@
+      }
+
+   static int32_t getNodeIntValue(TR::Node * node)
+      {
+      return node->getInt();
+      }
+
+   static int64_t getNodeLongValue(TR::Node * node)
+      {
+      return node->getLongInt();
+      }
+
+   static uint32_t getNodeFloatBits(TR::Node * node)
+      {
+      return node->getFloatBits();
+      }
+
+   static uint64_t getNodeDoubleBits(TR::Node * node)
+      {
+      return node->getDoubleBits();
+      }
+
+   static void setNodeByteValue(TR::Node * node, int8_t value)
+      {
+      node->setByte(value);
       }
 
+   static void setNodeShortValue(TR::Node * node, int16_t value)
+      {
+      node->setShortInt(value);
+      }
+
+   static void setNodeIntValue(TR::Node * node, int32_t value)
+      {
+      node->setInt(value);
+      }
+
+   static void setNodeLongValue(TR::Node * node, int64_t value)
+      {
+      node->setLongInt(value);
+      }
+
+   static void setNodeFloatBits(TR::Node * node, uint32_t bits)
+      {
+      node->setFloatBits(bits);
+      }
+
+   static void setNodeDoubleBits(TR::Node * node, uint64_t bits)
+      {
+      node->setDoubleBits(bits);
+      }
+   };
+
+template <typename T>
+struct BinaryOpSimplifier
+   {
+   TR::Simplifier * s;
+   T (* getNodeValue)(TR::Node *);
+   void (* setNodeValue)(TR::Node *, T);
+
+   /**
+    * Binary identity operation
+    *
+    * If the second child is a constant that represents an identity operand
+    * for the operation, replace this node with the first child.
+    */
+   inline TR::Node * tryToSimplifyIdentityOp(TR::Node * node, T identityValue)
+      {
+      auto secondChild = node->getSecondChild();
+      if (!secondChild || !secondChild->getOpCode().isLoadConst())
+         return NULL;
+
+      if (getNodeValue(secondChild) != identityValue)
+         return NULL;
+
+      return s->replaceNode(node, node->getFirstChild(), s->_curTree);
+      }
+
+   /**
+    * Binary identity or zero operation
+    *
+    * If the second child is a constant that represents an identity operand
+    * for the operation, replace this node with the first child.
+    * Binary zero operation
+    *
+    * If the second child is a constant that represents a zero operation,
+    * replace this node with the second child.
+    */
+   inline TR::Node * tryToSimplifyIdentityOrZeroOp(TR::Block * block, TR::Node * node, T identityValue, T zeroValue)
+      {
+      auto secondChild = node->getSecondChild();
+      if (!secondChild || !secondChild->getOpCode().isLoadConst())
+         return NULL;
+
+      auto value = getNodeValue(secondChild);
+      if (value == identityValue)
+         return s->replaceNodeWithChild(node, node->getFirstChild(), s->_curTree, block);
+
+      if (value != zeroValue)
+         return NULL;
+
+      auto transformMessage = "%sFound op with %s in node [" POINTER_PRINTF_FORMAT "]\n";
+      if (performTransformation(s->comp(), transformMessage, s->optDetailString(), node->getOpCode().getName(), node))
```

Comment:
This might seem picky, but I think most &mdash; if not all &mdash; calls to `performTransformation` have the message format string appear directly as an argument to the method.  It can make it easier to find the particular call to `performTransformation` with a regular expression, like `"performTransformation.*Found op"`

Reply:
squashed (-: